src/main/groovy/info/solidsoft/gradle/pitest/report/PitestReportAggregator.groovy

```diff
@@ -0,0 +1,111 @@
+
+import java.nio.file.Paths
+
+@CompileStatic
+class PitestReportAggregator {
+
+    private static final String MUTATION_FILE_NAME = "mutations.xml";
+    private static final String LINECOVERAGE_FILE_NAME = "linecoverage.xml";
+
+    void aggregate(Project root) {
+        Set<Project> subprojects = getSubprojectsWithPitest(root)
+        boolean rootContainPitest = isRootPitestConfigured(root)
+        if (rootContainPitest || !subprojects.isEmpty()) {
+            root.getLogger().info("Aggregating pitest reports")
+            ReportAggregator.Builder builder = ReportAggregator.builder()
+            if (rootContainPitest) {
+                addProjectFiles(builder, root)
+            }
+            subprojects.each { project ->
+                addProjectFiles(builder, project)
+            }
+
+            String outputDir = Paths.get(root.buildDir.absolutePath, "reports", "pitest").toFile().absolutePath
+
+            ReportAggregator aggregator = builder.resultOutputStrategy(new DirectoryResultOutputStrategy(
+                outputDir,
+                new UndatedReportDirCreationStrategy()))
+                .build()
+            aggregator.aggregateReport()
+        }
+    }
+
+    private static boolean isRootPitestConfigured(Project root) {
+        try {
+            return root.plugins.hasPlugin(PitestPlugin.PLUGIN_ID) &&
+                root.extensions.getByType(PitestPluginExtension) != null
+        } catch (UnknownDomainObjectException ignore) {
+            return false
+        }
+    }
+
+    private static Set<Project> getSubprojectsWithPitest(Project root) {
+        return root.subprojects.findAll { project -> project.plugins.hasPlugin(PitestPlugin.PLUGIN_ID) }
+    }
+
+    private static void addProjectFiles(ReportAggregator.Builder builder, Project project) {
+        PitestPluginExtension extension = project.extensions.getByType(PitestPluginExtension)
+        DirectoryProperty reportDir = extension.reportDir
+        PitestTask task = project.getTasks().getByName(PitestPlugin.PITEST_TASK_NAME) as PitestTask
+        addMutationsFile(builder, reportDir)
+        addLinecoverageFile(builder, reportDir)
+        addSourceCodeDirs(builder, task)
+        addCompiledCodeDirs(builder, task)
+    }
+
+    private static void addMutationsFile(ReportAggregator.Builder builder, DirectoryProperty reportDir) {
+        File file = getSingleFileFromDirectory(reportDir, MUTATION_FILE_NAME)
+        if (file != null) {
+            builder.addMutationResultsFile(file)
+        }
+    }
+
+    private static void addLinecoverageFile(ReportAggregator.Builder builder, DirectoryProperty reportDir) {
+        File file = getSingleFileFromDirectory(reportDir, LINECOVERAGE_FILE_NAME)
+        if (file != null) {
+            builder.addLineCoverageFile(file)
+        }
+    }
+
+    private static File getSingleFileFromDirectory(DirectoryProperty directory, String filename) {
```

Comment:
What do you think about returning a provider here, and treating it as `Optional` to perform the operation only if there is a real file?